workitem/workitem_repository_blackbox_test.go

```diff
@@ -322,3 +322,43 @@ func (s *workItemRepoBlackBoxTest) TestLookupIDByNamedSpaceAndNumberNotFound() {
 	require.NotNil(s.T(), err)
 	assert.IsType(s.T(), errors.NotFoundError{}, errs.Cause(err))
 }
+
+func (s *workItemRepoBlackBoxTest) TestLookupIDByNamedSpaceAndNumberStaleSpace() {
```

Comment:
One request I have is to make every new test function have this structure:

```go
func (s *workItemRepoBlackBoxTest) TestMETHODTOTEST() {
   s.T().Run("description of good or bad test, e.g. StaleSpace", func(t *testing.T){
      // Test your code here, even if you only have one case to test for the METHODTOTEST.
   })
}
```
This allows us to more clearly see what tests we cover for a particular method. I know I didn't say that in my earlier review but I think it is a good habit to get away from those longish test function names. WDYT?

Reply:
Yes, I agree with you @kwk 
And we can break main test function when there are much more sub-tests under 1 function.
e.g> 
`CreateIteration` vs `CreateChildIteration`: See both are `POST` calls but they should be covered in different main test functions in the same file.